EarlGrey/Core/GREYElementInteraction.m

```diff
@@ -232,13 +232,6 @@ - (instancetype)performAction:(id<GREYAction>)action error:(__strong NSError **)
       [defaultNotificationCenter postNotificationName:kGREYDidPerformActionNotification
                                                object:nil
                                              userInfo:actionUserInfo];
-      // If we encounter a failure and going to raise an exception, raise it right away before
```

Comment:
Seems like a refactoring is needed. I removed that commit from this PR so we can get the other commits in.